services/upload/src/main/java/app/coronawarn/server/services/federation/upload/runner/TestDataGeneration.java

```diff
@@ -67,17 +72,22 @@ public TestDataGeneration(UploadServiceConfig uploadServiceConfig,
     return keydata;
   }
 
-  private DiagnosisKey makeKeyFromTimestamp(long timestamp) {
-    return DiagnosisKey.builder().withKeyData(randomByteData())
+  private FederationUploadKey makeKeyFromTimestamp(long timestamp) {
+    return FederationUploadKey.from(DiagnosisKey.builder().withKeyData(randomByteData())
         .withRollingStartIntervalNumber(generateRollingStartIntervalNumber(timestamp))
-        .withTransmissionRiskLevel(2)
+        .withTransmissionRiskLevel(generateTransmissionRiskLevel())
         .withConsentToFederation(true)
         .withCountryCode("DE")
         .withDaysSinceOnsetOfSymptoms(1)
         .withSubmissionTimestamp(timestamp)
         .withVisitedCountries(List.of("FR", "DK"))
         .withReportType(ReportType.CONFIRMED_TEST)
-        .build();
+        .build());
+  }
+
+  private int generateTransmissionRiskLevel() {
+    return Math.toIntExact(
+        getRandomBetween(RiskLevel.RISK_LEVEL_LOWEST_VALUE, RiskLevel.RISK_LEVEL_HIGHEST_VALUE));
```

Comment:
Not sure whether it's an issue or not that TRL values like 2 or 4 might be produced. Considering that the german app does not submit such values. Probably for testing it's not a problem

Reply:
Agreed. Maybe we have to think this through in the future. But distribution does it like this also. DEV and INT are filled with keys that have the wrong TRL.